### PR TITLE
Some minor language changes

### PR DESCRIPTION
**Pull request by astridlyre**
I made a few minor language changes, with the hope that it improves the readability and tone.

### CODE REVIEW

File: docs/guides/core-internals.mdx [hunk 1]
```diff
@@ -4,12 +4,12 @@ description: A simplified version of the core implementation
...
 
-This guide is beneficial for those who want to understand the core implementation of the jotai. It is not meant to be a complete guide of the 
-core implementation but rather a simplified version. It is inspired by the collection of tweets by Daishi Kato([@dai_shi](https://twitter.com/dai_shi)).
+This guide is beneficial for those who want to understand the core implementation of jotai. It's not meant to be a complete example of the 
+core implementation but rather a simplified version. It's inspired by the collection of tweets by Daishi Kato([@dai_shi](https://twitter.com/dai_shi)).
 
 ## First Version
-Let's start with the easy one. An atom is just a function that will return some config. We are using WeakMap to map atom with their state.
-WeakMap doesn't keep the keys in memory. So if atom is garbage collected, its state will be garbage collected too. It helps to avoid memory leaks.
+Let's start with an easy example. An atom is just a function that will return some configured value. We are using WeakMap to map atom with their state.
```
Comment (sandren) [inline docs/guides/core-internals.mdx:11]: It would be more accurate to say `configuration object` here rather than `configured value`.

**Comment by sandren** [comment]
Thank you for helping with the docs!

Everything looks good except for that one comment. 🙂

File: docs/guides/core-internals.mdx (hunk 1 above)
Comment (astridlyre) [reply docs/guides/core-internals.mdx]: Sounds great! I've updated it. Thanks :smile:

**Review by dai-shi — APPROVED** [review]
@astridlyre This is very helpful! So much appreciated. Please feel free to work on other docs if you can.

**Pull request merged**